pkg/bff/api/v1/credentials.go

```diff
@@ -0,0 +1,120 @@
+package api
+
+import (
+	"encoding/json"
+	"os"
+	"time"
+)
+
+// Credentials provides a basic interface for loading an access token from Auth0 into
+// the BFF API Client. Credentials can be loaded from disk, generated, or directly
+// authenticated with Auth0 via a machine-to-machine token.
+//
+// NOTE: this is a fundamentally different mechanism than the GDS Admin API credentials
+// because the Admin API generates its own tokens, and Auth0 manages the BFF credentials.
+type Credentials interface {
+	AccessToken() (string, error)
+}
```

Comment:
@pdeziel this is one way I'm hoping to unblock SC-6366; by allowing us to directly use the BFF client for testing and injecting the access token either as a string or loaded from fixtures.